kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/test/TestContext.kt

```diff
@@ -44,3 +45,14 @@ interface TestContext : CoroutineScope {
     */
    suspend fun registerTestCase(nested: NestedTest)
 }
+
+/**
+ * Returns a new [TestContext] which uses the given [coroutineContext] with the other methods
+ * delegating to the receiver context.
+ */
+internal fun TestContext.withCoroutineContext(coroutineContext: CoroutineContext) = object : TestContext {
```

Comment:
For a large number of extensions the delegation chain will be quite long, consider a small optimization here:

```kotlin
private class TestContextWithCoroutineContext(
   val delegate: TestContext,
   override val coroutineContext: CoroutineContext
) : TestContext by delegate {
   override fun toString(): String = "TestCaseContext [$coroutineContext]"
}

/**
 * Returns a new [TestContext] which uses the given [coroutineContext] with the other methods
 * delegating to the receiver context.
 */
internal fun TestContext.withCoroutineContext(coroutineContext: CoroutineContext): TestContext =
   if (this is TestContextWithCoroutineContext) {
      TestContextWithCoroutineContext(delegate, coroutineContext)
   } else {
      TestContextWithCoroutineContext(this, coroutineContext)
   }
```

Reply:
Done.